README.md

```diff
@@ -1,116 +1,208 @@
+* **Description** Semistructured merge is unable to match **initialization blocks** (static and non-static), as it uses identifiers for this purpose and these elements have no identifiers, creating duplicates at the end. This handler tries to overcome this issue, matching nodes by content similarity.
+* **Behavior** If there's only one initialization block in all contributions, they're matched. Otherwise, for each initialization block in base, we search the most similar block in left and in right and merge them using textual merge.
 
-Where *mine*, *base*, *theirs* and *output* are filepaths.
-The attribute -o is optional, if omitted, *theirs* is used as the output file.
+### [Method and Constructor Renaming or Deletion Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/MethodAndConstructorRenamingAndDeletionHandler.java) 
+* **Description** Semistructured merge is unable to match **methods** or **constructors** that have been renamed (as their identifiers are now different) or deleted (as their nodes aren't created). This handler handles such cases, attempting to match nodes following other rules.
+* **Behavior** For each method or constructor renamed or deleted, we search left's nodes and right's nodes for a match. For this handler, is enough for one of the following conditions to be true for the match to happen:
+(1) have equal bodies; (2) have similar bodies and equal signatures except for the name; (3) one body is substring of the other. If none of the conditions are satisfied, we treat the node as deleted. Finally, for each triple, we apply a decision based on the type of the renaming or deletion:
+    * If both contributions renamed without body changes, we report a conflict if and only if they renamed to different signatures.
+    * If both contributions deleted or renamed with body changes, we immediately report a conflict if they renamed to different signatures. Otherwise, we merge their contents using textual merge.
+    * If one renamed without body changes and the other deleted renamed with body changes, we report a conflict if they renamed to different signatures or if they renamed to equal signatures but the latter included a reference to the renamed method/constructor.
 
-* Merging 3 directories:
+### [New Element Referencing Edited One Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/NewElementReferencingEditedOneHandler.java)
 
-  `java -jar pathto/jFSTMerge.jar -d "mine" "base" "theirs" -o "output"`
+* **Description** If a developer adds a **class member** that references an edited one inside its body by another developer, semistructured merge can change the semantic of the code since it doesn't care about the elements' bodies. Note that structured merge, on the other hand, deals with this naturally. This handler attempts to fix this issue.
```

Comment:
nioguem vai entender isto: If a developer adds a **class member** that references an edited one inside its body by another developer, semistructured merge can change the semantic of the code since it doesn't care about the elements' bodies. Note that structured merge, on the other hand, deals with this naturally.

foca em quando o handler e chamado, e o que exatamente ele verifica e faz.